src/renderer/actions/paths/linux.js

```diff
@@ -0,0 +1,71 @@
+const fs = require("fs");
+const path = require("path");
+import {remote} from "electron";
+const semverRcompare = require("semver/functions/rcompare");
+const semverValid = require("semver/functions/valid");
+
+/** @typedef {{basedir: string, dir: string, version: string}} VersionPath  */
+
+/**
+ * @param {string} basedir 
+ * @returns {VersionPath[]}
+ */
+const getDiscordVersionsForBaseDir = function(basedir) {
+    return fs.readdirSync(basedir).filter(
+                dir => 
+                    fs.lstatSync(path.join(basedir, dir)).isDirectory() && // filter for directories
+                    semverValid(dir.replace("app-","")) && // which have valid semver name (ignoring the app- prefix)
+                    fs.existsSync(path.join(basedir, dir, "modules", "discord_desktop_core")) // that have a modules/discord_desktop_core folder
+            ).map(
+                dir => ({
+                    basedir,
+                    dir,
+                    version: dir.replace("app-","")
+                })
+            );
+};
+
+// TODO: try some deduplication with the windows version
+/**
+ * @param {VersionPath[]} versions 
+ * @returns {string} path
+ */
+const getHighestVersion = function(versions) {
+    const highest = versions.sort((v1,v2)=>semverRcompare(v1.version,v2.version))[0];
+    return path.join(highest.basedir, highest.dir, "modules", "discord_desktop_core");
+};
+
+export const getDiscordPath = function(releaseChannel) {
+    const basedir = path.join(remote.app.getPath("appData"), releaseChannel.toLowerCase().replace(" ", ""));
+    if (!fs.existsSync(basedir)) return "";
+    
+    const detectedVersions = getDiscordVersionsForBaseDir(basedir);
+    console.debug(`Detected versions for channel "${releaseChannel}":`, detectedVersions);
+    
+    if (!detectedVersions.length) return "";
+
+    return getHighestVersion(detectedVersions);
```

Comment:
Why not get the highest version from the directory list rather than performing several `fs` checks before this point? It would also eliminate the need for the `map` and objectified return.